crates/mun_codegen/src/code_gen/symbols.rs

```diff
@@ -88,108 +103,277 @@ pub(super) fn gen_symbols(
         false,
     );
 
-    let method_info_type = context.opaque_struct_type("struct.MunFunctionInfo");
-    method_info_type.set_body(
+    // Construct the `MunFunctionSignature` type
+    let function_signature_type = context.opaque_struct_type("struct.MunFunctionSignature");
+    function_signature_type.set_body(
         &[
             str_type.into(),                                     // name
             type_info_type.ptr_type(AddressSpace::Const).into(), // arg_types
             type_info_type.ptr_type(AddressSpace::Const).into(), // return_type
+            context.i16_type().into(),                           // num_arg_types
+            privacy_type.into(),                                 // privacy
+        ],
+        false,
+    );
+
+    // Construct the `MunFunctionInfo` struct
```

Comment:
To reduce the number of lines of code in this function, maybe move every commented section into a separate function?

Reply:
I dont agree with this, I prefer to have the entire ABI struct definitions in one place. I could move it to a seperate module?

Reply:
I moved it to a seperate module.